### PR TITLE
Added Terraform to provision Synapse and Riot

### PR DESCRIPTION
**Pull request by ASoTNetworks**

### CODE REVIEW

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ - '172.16.0.0/12'
+ - '192.168.0.0/16'
+ - '100.64.0.0/10'
+ - '169.254.0.0/16'
+ - '::1/128'
+ - 'fe80::/64'
+ - 'fc00::/7'
+ ">> homeserver.yaml
+
+sed -i '/# Database configuration/,/event_cache_size: "10K"/d' homeserver.yam
```
[the remark this answers is not in the document]
Comment (benhylau) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: I didn't think we do this, do we?

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    args:
+        user: synapse_user
+        password: $DATAPASS
+        database: synapse
+        host: localhost
+        cp_min: 5
+        cp_max: 10
+
+# Number of events to cache in memory.
+event_cache_size: \"10K\"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: Not sure if we do this?

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+" >> homeserver.yaml
+echo "synctl_cache_factor: 0.02" >> homeserver.yaml
+
+# Install lxml
+pip install lxml
+
+# Create a startup script
+mkdir ~/bin/
+cp /tmp/matrix-server/synapse-startup.sh ~/bin/startup.sh
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: Why not name the dst the same as `synapse-startup.sh`? Less unnecessary mapping is easier to read

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: It may be useful to add `-e` switches to know when scripts failed

File: matrix-synapse-riot/terraform/matrix-server/psql-setup.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+DATAPASS=$1
+
+# Create user and database
+createuser synapse_user
+psql -c "alter user synapse_user with encrypted password '$DATAPASS'"
+psql -c "CREATE DATABASE synapse ENCODING 'UTF8' LC_COLLATE='C' LC_CTYPE='C' template=template0 OWNER synapse_user"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/psql-setup.sh]: In the README, we should describe how to _use_ this synapse admin user

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+########
+# Riot #
+########
+cd /var/www/
+wget https://github.com/vector-im/riot-web/releases/download/v$RIOT_VERSION/riot-v$RIOT_VERSION.tar.gz
+mkdir -p /var/www/chat.$DOMAIN_NAME/public
+tar xf riot-v$RIOT_VERSION.tar.gz -C /var/www/chat.$DOMAIN_NAME/public --strip-components 1
+cp /tmp/matrix-server/riot-config.json /var/www/chat.$DOMAIN_NAME/public/config.json
+sed -i -e "s/DOMAIN/$DOMAIN_NAME/g" /var/www/chat.$DOMAIN_NAME/public/config.json
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: I prefer using two underscores around macros to avoid unintended replacements `__DOMAIN_NAME__` and try to match them exactly as the var name that will eventually replace them to make code more readable

File: matrix-synapse-riot/terraform/matrix-server/riot-config.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    "features": {
+	"feature_groups": "labs",
+	"feature_pinning": "labs"
+    },
+    "default_federate": true,
+    "welcomePageUrl": "home.html",
+    "default_theme": "light",
+    "roomDirectory": {
+	"servers": [
+	    "DOMAIN",
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/riot-config.json]: Here

File: matrix-synapse-riot/terraform/matrix-server/riot-config.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{
+    "default_hs_url": "https://matrix.DOMAIN",
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/riot-config.json]: Here

File: matrix-synapse-riot/terraform/matrix-server/riot-config.json [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    "integrations_rest_url": "https://scalar.vector.im/api",
+    "integrations_jitsi_widget_url": "https://scalar.vector.im/api/widgets/jitsi.html",
+    "bug_report_endpoint_url": "https://riot.im/bugreports/submit",
+    "features": {
+	"feature_groups": "labs",
+	"feature_pinning": "labs"
+    },
+    "default_federate": true,
+    "welcomePageUrl": "home.html",
+    "default_theme": "light",
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/riot-config.json]: It it's up to me I'd default dark 😆

File: matrix-synapse-riot/terraform/matrix-server/nginx-chat [hunk 10]
```diff
@@ -0,0 +1,50 @@
+server {
+    listen 80;
+    listen [::]:80;
+    server_name chat.CHATDNS h.chat.CHATDNS;
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/nginx-chat]: May I suggest `s/CHATDNS/__DOMAIN_NAME__/g`?

File: matrix-synapse-riot/terraform/matrix-server/nginx-matrix [hunk 11]
```diff
@@ -0,0 +1,68 @@
+server {
+    listen 80;
+    listen [::]:80;
+    server_name matrix.CHATDNS h.matrix.CHATDNS;
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/nginx-matrix]: `__DOMAIN_NAME__` here too

File: matrix-synapse-riot/terraform/matrix-server/debian-50unattended-upgrades [hunk 12]
```diff
@@ -0,0 +1,105 @@
+// Unattended-Upgrade::Origins-Pattern controls which packages are
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/debian-50unattended-upgrades]: Curious where these files come from?

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 13]
```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env bash
+
+# Define DO API token
+DO_TOKEN=TOKENSTRING
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: `__DO_TOKEN__`

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+DO_DOMAIN=DODOMAIN
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: `__DO_DOMAIN__`

File: matrix-synapse-riot/terraform/main.tf [hunk 15]
```diff
@@ -0,0 +1,250 @@
+# Digital Ocean Tag for all server instances
+resource "digitalocean_tag" "matrix-home-service" {
+  name = "matrix-homeserver"
+}
+
+# Domain name managed by Digital Ocean
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: Why commented out?

File: matrix-synapse-riot/terraform/main.tf [hunk 16]
```diff
@@ -0,0 +1,250 @@
...
+#  #ip_address  = "${digitalocean_droplet.matrix-server.ipv4_address}"
+#  #ip6_address = "${digitalocean_droplet.matrix-server.ipv6_address}"
+#}
+
+# Matrix server Droplet
+resource "digitalocean_droplet" "matrix-server" {
+  image              = "debian-9-x64"
+  name               = "matrix.${file(var.domain_name)}"
+  region             = "tor1"
+  size               = "1gb"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: Why is this `1gb`? The deployed one is `2gb`

File: matrix-synapse-riot/terraform/main.tf [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+resource "digitalocean_record" "matrix-v6" {
+  domain = "${file(var.domain_name)}"
+  type   = "AAAA"
+  name   = "matrix"
+  value  = "${digitalocean_droplet.matrix-server.ipv6_address}"
+  ttl    = "86400"
+}
+resource "digitalocean_record" "matrix-caa" {
+  domain = "${file(var.domain_name)}"
+  type   = "CAA"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: This is for issuing wildcard cert to cjdns address?

File: matrix-synapse-riot/terraform/main.tf [hunk 18]
```diff
@@ -0,0 +1,250 @@
...
+resource "null_resource" "matrix-server" {
+  depends_on         = ["digitalocean_record.matrix"]
+  connection {
+    host             = "${digitalocean_droplet.matrix-server.ipv4_address}"
+    user             = "root"
+    type             = "ssh"
+    private_key      = "${file(var.pvt_key)}"
+    timeout          = "2m"
+  }
+  # Setup services such as NGINX, Let's Encrypt, Matrix, etc.
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: `Set up` two words

File: matrix-synapse-riot/terraform/main.tf [hunk 19]
```diff
@@ -0,0 +1,250 @@
...
+  provisioner "remote-exec" {
+    inline           = [
+      "/tmp/matrix-server/bootstrap-post-dns.sh ${file(var.domain_name)} ${file(var.do_token)}",
+    ]
+  }
+}
+
+# Setup CJDNS if selected
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: Just do a global string replace of `s/CJDNS/cjdns`

File: matrix-synapse-riot/terraform/main.tf [hunk 20]
```diff
@@ -0,0 +1,250 @@
...
+  count = "${var.cjdns != "0" ? 1 : 0}"
+  depends_on         = ["null_resource.matrix-server"]
+  connection {
+    host             = "${digitalocean_droplet.matrix-server.ipv4_address}"
+    user             = "root"
+    type             = "ssh"
+    private_key      = "${file(var.pvt_key)}"
+    timeout          = "2m"
+  }
+  # Setup CJDNS
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: Maybe also a global `setup` -> `set up` 😆

File: matrix-synapse-riot/terraform/main.tf [hunk 21]
```diff
@@ -0,0 +1,250 @@
...
+  domain     = "${file(var.domain_name)}"
+  type       = "CAA"
+  name       = "h.matrix"
+  flags      = 0
+  tag        = "issue"
+  value      = "letsencrypt.org."
+  ttl        = "86400"
+}
+
+# Get cert from Let's Encrypt
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/main.tf]: Perhaps mention the use of dehydrated to do this somewhere?

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-cleanup.sh [hunk 22]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+# Disable root login in sshd_config
+sed -i -e "s/PermitRootLogin yes/PermitRootLogin no/g" /etc/ssh/sshd_config
+
+# Clean up the tmp directory
+shred -u /tmp/matrix-server/*
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-cleanup.sh]: Does this get rid of the folder?

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+su -c "/tmp/matrix-server/synapse-setup.sh $DOMAIN_NAME $DATAPASS" synapse
+
+########
+# Riot #
+########
+cd /var/www/
+wget https://github.com/vector-im/riot-web/releases/download/v$RIOT_VERSION/riot-v$RIOT_VERSION.tar.gz
+mkdir -p /var/www/chat.$DOMAIN_NAME/public
+tar xf riot-v$RIOT_VERSION.tar.gz -C /var/www/chat.$DOMAIN_NAME/public --strip-components 1
+cp /tmp/matrix-server/riot-config.json /var/www/chat.$DOMAIN_NAME/public/config.json
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: First sed replace in the tmp file. Copy the file after everything is done. This avoid intermediate states.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh [hunk 24]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: I think all files could use a `set -e`

File: matrix-synapse-riot/terraform/main.tf (hunk 16 above)
Comment (benhylau) [reply matrix-synapse-riot/terraform/main.tf]: Also enable https://www.terraform.io/docs/providers/do/d/droplet.html#backups

File: matrix-synapse-riot/terraform/matrix-server/nginx-chat [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+    ssl_certificate /etc/nginx/ssl/chat.CHATDNS/fullchain.pem;
+    ssl_certificate_key /etc/nginx/ssl/chat.CHATDNS/privkey.pem;
+    ssl_trusted_certificate /etc/nginx/ssl/chat.CHATDNS/fullchain.pem;
+
+    ssl_session_timeout 1d;
+    ssl_session_cache shared:SSL:50m;
+
+    # Diffie-Hellman parameter for DHE ciphersuites, recommended 4096 bits
+    # Generate with: openssl dhparam -out /etc/nginx/dhparam.pem 4096
+    ssl_dhparam /etc/ssl/certs/dhparam.pem;
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/nginx-chat]: This differs from the path in above comment

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh [hunk 26]
```diff
@@ -0,0 +1,103 @@
...
+chmod +x /tmp/matrix-server/dehydrated-hooks /opt/dehydrated-$DEHYDRATED_VERSION/hooks.sh
+ln -s /opt/dehydrated-$DEHYDRATED_VERSION /opt/dehydrated
+cd /opt/dehydrated
+sed -i -e "s/TOKENSTRING/$DO_TOKEN/g" hooks.sh
+sed -i -e "s/DODOMAIN/$DOMAIN_NAME/g" hooks.sh
+
+# Register to Let's Encrypt
+./dehydrated --register --accept-terms
+
+cp /tmp/matrix-server/dehydrated-renew.sh /usr/local/sbin/dehydrated-renew.sh
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: `chmod +x dehydrated-renew.sh`?

File: matrix-synapse-riot/terraform/matrix-server/nginx-matrix [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+    return 301 https://$host$request_uri;
+    
+    location ~ /.well-known {
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/nginx-matrix]: Remove all `.well-known` blocks since they are not used in the way we verify

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+    # - TOKEN_VALUE
+    #   The token value that needs to be served for validation. For DNS
+    #   validation, this is what you want to put in the _acme-challenge
+    #   TXT record. For HTTP validation it is the value that is expected
+    #   be found in the $TOKEN_FILENAME file.
+
+    # Simple example: Use nsupdate with local named
+    # printf 'server 127.0.0.1\nupdate add _acme-challenge.%s 300 IN TXT "%s"\nsend\n' "${DOMAIN}" "${TOKEN_VALUE}" | nsupdate -k /var/run/named/session.key
+    DO_HOSTNAME=$(echo $DOMAIN | sed -e "s/$DO_DOMAIN//" | sed 's/.$//')
+    echo "Issueing channenge for $DO_HOSTNAME Lets Encrypt domain is $DOMAIN Challenge is $TOKEN_VALUE"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: Spellings `Issuing` and `challenge`. There are a couple more spelling mistakes throughout the messages, please run a check.

Comment (benhylau) [reply matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: The format of the UI logging can use some improvement.

File: matrix-synapse-riot/terraform/matrix-server/ip6tables-rules [hunk 29]
```diff
@@ -0,0 +1,14 @@
+# Generated by ip6tables-save v1.6.0 on Sat Nov 17 04:28:07 2018
+*filter
+:INPUT DROP [0:0]
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [118:17738]
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/ip6tables-rules]: Change all these numbers to `[0:0]` as they are kind of arbitrary.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh [hunk 30]
```diff
@@ -0,0 +1,103 @@
...
+
+# Configure auto-renewals
+sh -c '(crontab -l 2>/dev/null; echo "@weekly /usr/local/sbin/dehydrated-renew.sh") | crontab -'
+
+############
+# iptables #
+############
+cp /tmp/matrix-server/ip6tables-rules /etc/iptables/rules.v6
+cp /tmp/matrix-server/iptables-rules /etc/iptables/rules.v4
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: Don't you need to:

```
iptables-apply rules.v4
ip6tables-apply rules.v6
```

To apply the rules in these files?

File: matrix-synapse-riot/terraform/matrix-server/ip6tables-rules [hunk 31]
```diff
@@ -0,0 +1,14 @@
...
+-A INPUT -p tcp -m tcp --dport 22 -j ACCEPT
+-A INPUT -p tcp -m tcp --dport 443 -j ACCEPT
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/ip6tables-rules]: Arrange lines in order?

File: matrix-synapse-riot/terraform/matrix-server/iptables-rules [hunk 32]
```diff
@@ -0,0 +1,14 @@
+# Generated by iptables-save v1.6.0 on Sat Nov 17 04:28:03 2018
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/iptables-rules]: Same as v6 file

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh [hunk 33]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh]: Should this file be named `dehydrated-bootstrap` for consistency with `dehydrated-renew`?

They are actually the same file, why not just reference one copy?

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh [hunk 34]
```diff
@@ -0,0 +1,22 @@
...
+
+# Get certs will try a maximum of 3 fail attempts
+cd /opt/dehydrated/
+count=0
+false
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh]: Why is this needed with the `$?` check?

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh [hunk 35]
```diff
@@ -0,0 +1,22 @@
...
+		echo "SLEEPING FOR 60 SECONDS BEFORE RETRY"
+		sleep 60
+	fi
+	let count=count+1
+	echo "SSL cert try: $count"
+	./dehydrated -c -f config
+done
+if [ $? -ne 0 ]
+then
+	echo "SSL CERT FAILED!!!"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh]: Maybe `SSL certificate request failed` to give better context?

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh [hunk 36]
```diff
@@ -0,0 +1,22 @@
...
+
+# Get certs will try a maximum of 3 fail attempts
+cd /opt/dehydrated/
+count=0
+false
+while [ $? -ne 0 ] && [ $count -lt 3 ]
+do
+	if [ $count -ne 0 ]
+	then
+		echo "SLEEPING FOR 60 SECONDS BEFORE RETRY"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh]: Not a fan of the all caps on messages

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-config [hunk 37]
```diff
@@ -0,0 +1,117 @@
...
+#BASEDIR=$SCRIPTDIR
+
+# File containing the list of domains to request certificates for (default: $BASEDIR/domains.txt)
+#DOMAINS_TXT="${BASEDIR}/domains.txt"
+
+# Output directory for generated certificates
+#CERTDIR="${BASEDIR}/certs"
+
+# Directory for account keys and registration information
+ACCOUNTDIR="${BASEDIR}/accounts"
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-config]: Why do you uncomment this and `HOOK`, but not `DOMAINS_TXT` and `CERTDIR `? Assuming the intention is to lock down config explicitly.

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 38]
```diff
@@ -0,0 +1,214 @@
...
+    # - TIMESTAMP
+    #   Timestamp when the specified certificate was created.
+
+    # Simple example: Copy file to nginx config
+    if [ ! -d /etc/nginx/ssl/$DOMAIN/ ]
+    then
+    	mkdir -m  700 /etc/nginx/ssl/$DOMAIN/
+    fi
+    cp "${KEYFILE}" "${FULLCHAINFILE}" /etc/nginx/ssl/$DOMAIN/ ; chown -R root:root /etc/nginx/ssl/
+    # systemctl reload nginx
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: I think it is better to keep their example unmodified and all together, as in other parts:

```
    # Simple example: Copy file to nginx config
    # cp "${OCSPFILE}" /etc/nginx/ssl/; chown -R nginx: /etc/nginx/ssl
    # systemctl reload nginx
```

Then put our custom stuff. Looking at this now I am quite confused.

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 39]
```diff
@@ -0,0 +1,214 @@
...
+    #   The path of the file containing the full certificate chain.
+    # - CHAINFILE
+    #   The path of the file containing the intermediate certificate(s).
+    # - TIMESTAMP
+    #   Timestamp when the specified certificate was created.
+
+    # Simple example: Copy file to nginx config
+    if [ ! -d /etc/nginx/ssl/$DOMAIN/ ]
+    then
+    	mkdir -m  700 /etc/nginx/ssl/$DOMAIN/
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: There's an extra space

File: matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks [hunk 40]
```diff
@@ -0,0 +1,214 @@
...
+    fi
+    cp "${KEYFILE}" "${FULLCHAINFILE}" /etc/nginx/ssl/$DOMAIN/ ; chown -R root:root /etc/nginx/ssl/
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks]: - I think you are missing `$DOMAIN` at the end
- The `chown` line is better as a next line

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: > I didn't think we do this, do we?

That is to remove the default sqlite configuration and we write our own below.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 2 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: That get removed from the sed above and we add it back.

File: matrix-synapse-riot/terraform/matrix-server/debian-50unattended-upgrades (hunk 12 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/debian-50unattended-upgrades]: That is to enable auto apt update / upgrade on Debian.

File: matrix-synapse-riot/terraform/main.tf (hunk 15 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/main.tf]: Since we already have the domain already on DO it will fail to create it. So in the README I said it is required to have the domain already on DO DNS.

File: matrix-synapse-riot/terraform/main.tf (hunk 17 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/main.tf]: That is to set the Certification Authority Authorization to Let's Encrypt so that browsers will not trust another cert from a different CA. It is added because sometimes dehydrated will fail saying it cannot lookup the CAA record with SERVFAIL error.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh (hunk 26 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: I have chmod +x /tmp/matrix-server/*.sh in main.tf so all .sh files will have +x already.

**Comment by benhylau** [comment]
@ASoTNetworks most comments are still unaddressed, please let me know when ready for review. I also submitted a PR to improve README text and format consistency to your repo.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (benhylau) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: I mean, I thought we use default size, no?

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 2 above)
Comment (benhylau) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: Yes but I thought we use default.

File: matrix-synapse-riot/terraform/main.tf (hunk 20 above)
Comment (benhylau) [reply matrix-synapse-riot/terraform/main.tf]: I think some are still incorrect, do a global search?

File: matrix-synapse-riot/terraform/matrix-server/iptables-rules [hunk 41]
```diff
@@ -0,0 +1,14 @@
...
+*filter
+:INPUT DROP [16:1178]
```
Comment (benhylau) [inline matrix-synapse-riot/terraform/matrix-server/iptables-rules]: Zero this one.

File: matrix-synapse-riot/terraform/.keys/ipv6-cjdns [hunk 42]
```diff
@@ -0,0 +1 @@
+null
```
[the remark this answers is not in the document]
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/.keys/ipv6-cjdns]: This just needs to non-empty or terraform errors out.

File: matrix-synapse-riot/terraform/matrix-server/synapse-startup.sh [hunk 43]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+sleep 10
```
[the remark this answers is not in the document]
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-startup.sh]: It is safe I just needed a little delay for networking to be fully up before starting.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh [hunk 44]
```diff
@@ -0,0 +1,62 @@
...
+
+SYNAPSE_VERSION=0.33.9
```
[the remark this answers is not in the document]
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: The memory leak was something to do with running an newer version of python on the previous system.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: That is the default size but I have to re-add that line back to the config file as the above sed command removed it.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 2 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: That is the default size but I have to re-add that line back to the config file as the above sed command removed it.

File: matrix-synapse-riot/terraform/main.tf (hunk 20 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/main.tf]: Think I have replaced them all.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-cleanup.sh (hunk 22 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/bootstrap-cleanup.sh]: added rm -rf  /tmp/matrix-server/ under that.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh (hunk 30 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh]: Added:
```
ip6tables-restore /etc/iptables/rules.v6
iptables-restore /etc/iptables/rules.v4
```
instead of iptables-apply because it needs to be interactive.

File: matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh (hunk 33 above)
Comment (ASoTNetworks) [reply matrix-synapse-riot/terraform/matrix-server/bootstrap-dehydrated.sh]: I am keeping it named bootstrap-dehydrated.sh because its being called from main.tf and now there is only one copy of the file.

File: matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh (hunk 2 above)
Comment (benhylau) [reply matrix-synapse-riot/terraform/matrix-server/synapse-setup.sh]: I mean, why did we remove it in the first place?
